### ISSUE TITLE
Incorrect order: attempt to delete resource is in use

### ISSUE THREAD

**Issue by Z9n2JktHlZDmlhSvqc9X2MmL3BwQG7tk** [terraform-providers/terraform-provider-aws#4537]
I have faced with incorrect operations order: long lasting attempt to delete a cert which is in use by load balancer listener with no luck.
Tried to use `lifecycle { create_before_destroy = "true" }` in `aws_acm_certificate` and `depends_on` in `aws_lb_listener`, but it didn't help.

### Terraform Version
Terraform v0.11.7
provider.aws v1.17.0

### Affected Resource(s)
+ aws_acm_certificate
+ aws_lb_listener

### Terraform Configuration Files
```hcl
resource "aws_acm_certificate" "cert" {
  domain_name                 = "${var.domain_name}"
  subject_alternative_names   = ["www.${var.domain_name}"]
  validation_method           = "DNS"
}

resource "aws_acm_certificate_validation" "cert" {
  certificate_arn             = "${aws_acm_certificate.cert.arn}"
  validation_record_fqdns     = ["${aws_route53_record.cert_validation.fqdn}"]
}

resource "aws_lb_listener" "front_https" {
  certificate_arn            = "${aws_acm_certificate_validation.cert.certificate_arn}"
  ........
    
# depends_on                 = ["aws_acm_certificate.cert", "aws_acm_certificate_validation.cert"]
}
```

### Expected Behavior

Destruction of load balancer listener, creation of new cert and creation of new listener.
OR
New cert creation, listener update and old cert destruction (in `create_before_destroy` mode).

### Actual Behavior

Long attempt to delete cert.

### Steps to Reproduce
`terraform plan`

```
Terraform will perform the following actions:

-/+ aws_acm_certificate.cert (deposed) (new resource required)
      id:                                "arn:aws:acm:eu-west-3:766597584338:certificate/9b99d953-25fd-46ab-ada4-463d38c47898" => <computed> (forces new resource)
      arn:                               "arn:aws:acm:eu-west-3:766597584338:certificate/9b99d953-25fd-46ab-ada4-463d38c47898" => <computed>
      domain_name:                       "domain.com" => "domain.com"
      domain_validation_options.#:       "1" => <computed>
      subject_alternative_names.#:       "0" => "1" (forces new resource)
      subject_alternative_names.0:       "" => "www.domain.com" (forces new resource)
      validation_emails.#:               "0" => <computed>
      validation_method:                 "DNS" => "DNS"

  + aws_acm_certificate_validation.cert
      id:                                <computed>
      certificate_arn:                   "${aws_acm_certificate.cert.arn}"
      validation_record_fqdns.#:         "1"
      validation_record_fqdns.523129132: "_9ed9fa5d73c4gw676hsgn032xstw2b36.domain.com"

  ~ aws_lb_listener.front_https
      certificate_arn:                   "arn:aws:acm:eu-west-3:766597584338:certificate/9b99d953-25fd-46ab-ada4-463d38c47898" => "${coalesce(aws_acm_certificate_validation.cert.certificate_arn, aws_acm_certificate.cert.arn)}"

Plan: 2 to add, 1 to change, 1 to destroy.

```
\
\
`terraform apply`

```
aws_acm_certificate.cert: Destroying... (ID: arn:aws:acm:eu-west-3:766597584338:cert...e/9b99d953-25fd-46ab-ada4-463d38c47898)
aws_acm_certificate.cert: Still destroying... (ID: arn:aws:acm:eu-west-3:766597584338:cert...e/9b99d953-25fd-46ab-ada4-463d38c47898, 10s elapsed)
........
Error: Error applying plan:

1 error(s) occurred:

* aws_acm_certificate.cert (destroy): 1 error(s) occurred:

* aws_acm_certificate.cert: Error deleting certificate: ResourceInUseException: Certificate arn:aws:acm:eu-west-3:766597584338:certificate/9b99d953-25fd-46ab-ada4-463d38c47898 in account 766597584338 is in use.
        status code: 400, request id: 5f292694-57a6-11e8-823f-f96a122c64fc
```

**Comment by kitsunde** [comment]
I essentially had this issue in https://github.com/terraform-providers/terraform-provider-aws/issues/4712. However `create_before_destroy` does work for me. I'm using a newer versions of terraform and the AWS provider.

If you upgrade and set:

```hcl
lifecycle {
  create_before_destroy = true
}
```

On only the `aws_acm_certificate` it might work for you now.

I tried reproduce your issue with `create_before_destroy` and `aws_lb_listener` specifically. Using:

```hcl
provider "aws" {
  region = "us-east-1"
}

variable "domain" {}

variable "zone_name" {}

resource "aws_acm_certificate" "cert" {
  domain_name       = "${var.domain}"
  validation_method = "DNS"

  lifecycle {
    create_before_destroy = true
  }
}

data "aws_route53_zone" "zone" {
  name = "uatdomains.com."
}

resource "aws_route53_record" "records" {
  name = "${lookup(aws_acm_certificate.cert.domain_validation_options[0], "resource_record_name")}"
  type = "${lookup(aws_acm_certificate.cert.domain_validation_options[0], "resource_record_type")}"

  zone_id = "${data.aws_route53_zone.zone.id}"

  records = [
    "${lookup(aws_acm_certificate.cert.domain_validation_options[0], "resource_record_value")}",
  ]

  ttl = 60
}

resource "aws_acm_certificate_validation" "cert_validation" {
  certificate_arn = "${aws_acm_certificate.cert.arn}"

  validation_record_fqdns = [
    "${aws_route53_record.records.*.fqdn}",
  ]
}

resource "aws_lb" "test" {
  subnets = ["${aws_subnet.public.*.id}"]
}

resource "aws_vpc" "test" {
  cidr_block = "10.0.1.0/24"
}

resource "aws_internet_gateway" "gw" {
  vpc_id = "${aws_vpc.test.id}"
}

resource "aws_subnet" "public" {
  count             = 2
  availability_zone = "${element(list("us-east-1a", "us-east-1b"), count.index)}"
  cidr_block        = "${cidrsubnet(aws_vpc.test.cidr_block, 2, count.index)}"
  vpc_id            = "${aws_vpc.test.id}"
}

resource "aws_lb_target_group" "front_end" {
  port     = 1
  protocol = "HTTPS"
  vpc_id   = "${aws_vpc.test.id}"
}

resource "aws_lb_listener" "test" {
  certificate_arn = "${aws_acm_certificate_validation.cert_validation.certificate_arn}"
  protocol        = "HTTPS"

  lifecycle {
    create_before_destroy = true
  }

  "default_action" {
    target_group_arn = "${aws_lb_target_group.front_end.arn}"
    type             = "forward"
  }

  load_balancer_arn = "${aws_lb.test.arn}"
  port              = 1
}
```

Running:

```
terraform apply -var=zone_name=uatdomains.com. -var=domain=test123456.uatdomains.com
```

and then

```
terraform apply -var=zone_name=uatdomains.com. -var=domain=test1234567.uatdomains.com
```

It successfully replaced the certificate.

**Comment by bflad** [comment]
The documentation update to include the `create_before_destroy` note has been merged via #4925 and will release with version 1.25.0 of the AWS provider, likely middle of this week.
